### PR TITLE
Scrum 12 genres

### PR DESCRIPTION
**Pull request by san-ras**
Genre endpoint added

### CODE REVIEW

File: MovieGPT-api/src/main/java/lt/techin/group/project/exception/GenreNotFoundException.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 @ResponseStatus(HttpStatus.NOT_FOUND)
```
Comment (Stalkakuma) [inline MovieGPT-api/src/main/java/lt/techin/group/project/exception/GenreNotFoundException.java:6]: for duplicated data we can use HttpStatus.CONFLICT as a response, for more accurate status.

Comment (Stalkakuma) [reply MovieGPT-api/src/main/java/lt/techin/group/project/exception/GenreNotFoundException.java:6]: We can also add:
 @ExceptionHandler(GenreAlreadyExistsException.class)
    public ResponseEntity<ErrorResponse> handleGenreAlreadyExistsException(GenreAlreadyExistsException ex) {
        ErrorResponse errorResponse = new ErrorResponse(
                HttpStatus.CONFLICT.value(),
                "CONFLICT",
                ex.getMessage()
        );
        return ResponseEntity.status(HttpStatus.CONFLICT).body(errorResponse);
    }
    
   to GlobalExceptionHandler to catch this specific exception

Comment (san-ras) [reply MovieGPT-api/src/main/java/lt/techin/group/project/exception/GenreNotFoundException.java:6]: Fixed, pushed new code

**Review by Stalkakuma — APPROVED** [review]
Nice work, appreciate the tests!

**Pull request merged**